File: unidad2/BankAccount.java
public class BankAccount {
    private long accountNumber;
    private double amount;
    private String type;
    private boolean status = true;

    public BankAccount(long accountNumber, String type) {
        if(validType(type)){
            this.accountNumber = accountNumber;
            this.type = type;
        }
        else{
            this.status = false;
        }
            
    }

    public long getAccountNumber() {
        return accountNumber;
    }

    public String getType() {
        return type;
    }

    public double getAmount() {
        return amount;
    }

    public boolean isStatus() {
        return status;
    }

    private boolean validType(String type){
        switch(type){
            case "A":
            case "B":
            case "C":
                return true;

            default:
                return false;
        }
    }

    public void agregarDinero(double cargo){
        switch(type){
            case "A":
                agregarDineroA(cargo);
                System.out.println("Money added succesfully");
            break;

            case "B":
                agregarDineroB(cargo);
                System.out.println("Money added succesfully");
            break;

            case "C":
                agregarDineroC(cargo);
                System.out.println("Money added succesfully");
            break;   
        }
    }

    public void retirarDinero(double retiro){
        switch(type){
            case "A":
                retirarDineroA(retiro);
                System.out.println("Money taken succesfully");
            break;

            case "B":
                retirarDineroB(retiro);
                System.out.println("Money taken succesfully");
            break;

            case "C":
                retirarDineroC(retiro);
                System.out.println("Money taken succesfully");
            break; 
        }
    }



    private void agregarDineroA(double cargo){
        if(amount + cargo >= 50000){
            System.out.println("\nHa excedido la cantidad máxima de este tipo de cuenta");
        }
        else if(cargo < 0){
            System.out.println("\nNo puede realizar un cargo negativo");
        }
        else{
            amount += cargo;
        }
    }


    private void agregarDineroB(double cargo){
        if(amount + cargo >= 100000){
            System.out.println("\nHa excedido la cantidad máxima de este tipo de cuenta");
        }
        else if(cargo < 0){
            System.out.println("\nNo puede realizar un cargo negativo");
        }
        else{
            amount += cargo;
        }
    }


    private void agregarDineroC(double cargo){
        if(cargo < 0){
            System.out.println("\nNo puede realizar un cargo negativo");
        }
        else{
            amount += cargo;
        }
    }

    private void retirarDineroA(double retiro){
        if(amount < 1000){
            System.out.println("\nSaldo insuficiente para retirar");
        }
        else if(amount < retiro){
            System.out.println("\nEl dinero a retirar es mayor al saldo disponible");
        }
        else{
            amount -= retiro;
        }               
    }

    private void retirarDineroB(double retiro){
        if(amount < 5000){
            System.out.println("\nSaldo insuficiente para retirar");
        }
        else if(amount < retiro){
            System.out.println("\nEl dinero a retirar es mayor al saldo disponible");
        }
        else{
            amount -= retiro;
        } 
    }

    private void retirarDineroC(double retiro){
        if(amount < 10000){
            System.out.println("\nSaldo insuficiente para retirar");
        }
        else if(amount < retiro){
            System.out.println("\nEl dinero a retirar es mayor al saldo disponible");
        }
        else{
            amount -= retiro;
        }
    }



    
}

File: unidad2/BankAccount.py
class BankAccount:          ##CLASE LISTA
    __account_number = 0
    __amount = 0.0
    __type = ''
    __status = True

    def __init__(self, account_number, type):
        if(self._valid_type(type)):
            self.__account_number = account_number
            self.__type = type
        else:
            self.__status = False

    def get_account_number(self):
        return self.__account_number
    
    def get_account_type(self):
        return self.__type
    
    def get_amount(self):
        return self.__amount
    
    def is_status(self):
        return self.__status
    
    def _valid_type(self, type):
        if type == 'A' or type == 'B' or type == 'C':
            return True
        else:
            return False
            
    def agregar_dinero(self, cargo):
        if self.__type == 'A':
            self.agregar_dineroA(cargo)
            print('Money added succesfully')
        elif self.__type == 'B':
            self.agregar_dineroB(cargo)
            print('Money added succesfully')
        elif self.__type == 'C':
            self.agregar_dineroC(cargo)
            print('Money added succesfully')


    def retirar_dinero(self, retiro):
        if self.__type == 'A':
            self.retirar_dineroA(retiro)
            print('Money taken succesfully')
        elif self.__type == 'B':
            self.retirar_dineroB(retiro)
            print('Money taken succesfully')
        elif self.__type == 'C':
            self.retirar_dineroC(retiro)
            print('Money taken succesfully')



    def agregar_dineroA(self, cargo):
        if self.__amount + cargo >= 50000:
            print('\nHa excedido la cantidad máxima de este tipo de cuenta')
        elif cargo < 0:
            print('\nNo puede realizar un cargo negativo')
        else:
            self.__amount += cargo

    def agregar_dineroB(self, cargo):
        if self.__amount + cargo >= 100000:
            print('\nHa excedido la cantidad máxima de este tipo de cuenta')
        elif cargo < 0:
            print('\nNo puede realizar un cargo negativo')
        else:
            self.__amount += cargo

    def agregar_dineroC(self, cargo):
        if cargo < 0:
            print('\nNo puede realizar un cargo negativo')
        else:
            self.__amount += cargo


    def retirar_dineroA(self, retiro):
        if self.__amount < 1000:
            print('\nSaldo insuficiente para retirar')
        elif self.__amount < retiro:
            print('\nEl dinero a retirar es mayor al saldo disponible')
        else:
            self.__amount -= retiro

    def retirar_dineroB(self, retiro):
        if self.__amount < 5000:
            print('\nSaldo insuficiente para retirar')
        elif self.__amount < retiro:
            print('\nEl dinero a retirar es mayor al saldo disponible')
        else:
            self.__amount -= retiro

    def retirar_dineroC(self, retiro):
        if self.__amount < 10000:
            print('\nSaldo insuficiente para retirar')
        elif self.__amount < retiro :
            print('\nEl dinero a retirar es mayor al saldo disponible')
        else:
            self.__amount -= retiro
 